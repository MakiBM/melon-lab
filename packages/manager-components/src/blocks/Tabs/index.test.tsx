import React from 'react';
import { TabContent, Tabs } from './index';

describe('Tabs', () => {
  const defaultElement = (
    <Tabs>
      <TabContent>Tab 0</TabContent>
      <TabContent>Tab 1</TabContent>
    </Tabs>
  );

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be selectable by class "tabs"', () => {
    expect(shallow(defaultElement).is('.tabs')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(defaultElement).find('.tabs').length).toBe(1);
  });

  xit('should render to static HTML', () => {
    expect(render(defaultElement).text()).toEqual('Tabs');
  });
});
