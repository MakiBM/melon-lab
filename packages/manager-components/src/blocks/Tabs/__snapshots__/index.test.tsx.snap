// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly 1`] = `
<div
  className="jsx-__scopedHash tabs"
  role="tablist"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash tabs__nav"
  >
    <div
      aria-selected={false}
      className="tabs__tab"
      key="0"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    />
    <div
      aria-selected={false}
      className="tabs__tab"
      key="1"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    />
  </div>
  <div
    className="tabs__content"
    key="0"
  >
    <TabContent>
      Tab 0
    </TabContent>
  </div>
  <div
    className="tabs__content"
    key="1"
  >
    <TabContent>
      Tab 1
    </TabContent>
  </div>
</div>
`;
